packages/context/src/__tests__/acceptance/class-level-bindings.acceptance.ts

```diff
@@ -145,34 +148,181 @@ describe('Context bindings - Injecting dependencies of classes', () => {
+      expect(store.hash).to.equal('456');
+    });
+
+    it('injects method parameters from a child context', async () => {
+      class Store {
+        private name = 'my-store';
+
+        static getHash(@inject('hash') hash: string) {
+          return hash;
+        }
+
+        getHashWithName(@inject('hash') hash: string) {
+          return `${hash}: ${this.name}`;
+        }
+      }
+
+      givenRequestContext(Store);
+
+      let value = await invokeMethod(Store, 'getHash', requestCtx);
+      expect(value).to.equal('123');
+
+      const store = await requestCtx.get<Store>('store');
+      value = await invokeMethod(store, 'getHashWithName', requestCtx);
+      expect(value).to.equal('123: my-store');
+
+      // bind the value to to ctx
+      ctx.bind('hash').to('456');
+
+      value = await invokeMethod(Store, 'getHash', ctx);
+      expect(value).to.equal('456');
+
+      value = await invokeMethod(store, 'getHashWithName', ctx);
+      expect(value).to.equal('456: my-store');
+    });
+
+    it('injects only bindings from the owner context with @inject.tag', async () => {
+      class Store {
+        constructor(@inject.tag('feature') public features: string[]) {}
+      }
+
+      givenRequestContext(Store);
+
+      ctx
+        .bind('features.security')
+        .to('security')
+        .tag('feature');
+      requestCtx
+        .bind('features.debug')
+        .to('debug')
+        .tag('feature');
+
+      const store = await requestCtx.get<Store>('store');
+      // For singleton bindings, the injected tagged bindings will be only from
+      // the owner context (`ctx`) instead of the current one (`requestCtx`)
+      // used to resolve the bound `Store`
+      expect(store.features).to.eql(['security']);
+    });
+
+    it('injects owner context with @inject.context', async () => {
+      class Store {
+        constructor(@inject.context() public context: Context) {}
+      }
+
+      givenRequestContext(Store);
+
+      const store = await requestCtx.get<Store>('store');
+      // For singleton bindings, the injected context will be the owner context
+      // (`ctx`) instead of the current one (`requestCtx`) used to resolve the
+      // bound `Store`
+      expect(store.context).to.equal(ctx);
+    });
+
+    it('injects setter of owner context with @inject.setter', async () => {
+      class Store {
+        constructor(@inject.setter('hash') public setHash: Setter<string>) {}
+      }
+
+      givenRequestContext(Store);
+
+      const store = await requestCtx.get<Store>('store');
+      store.setHash('456');
+
+      // For singleton bindings, the injected setter will set value to the owner
+      // context (`ctx`) instead of the current one (`requestCtx`) used to
+      // resolve the bound `Store`
+      expect(await ctx.get<string>('hash')).to.equal('456');
+      expect(await requestCtx.get<string>('hash')).to.equal('123');
+    });
+
+    function givenRequestContext(storeClass: Constructor<unknown>) {
```

Comment:
The name does not tell the whole store about what the helper does. I am proposing the following improvement:

```suggestion
    function givenRequestContextWithHashAndStore(storeClass: Constructor<unknown>) {
```